pyquil/api/_qvm.py

```diff
@@ -510,18 +510,16 @@ def run(self):
             quil_program = apply_noise_model(quil_program, self.noise_model)
 
         quil_program = self.augment_program_with_memory_values(quil_program)
-        try:
-            self._bitstrings = self.connection._qvm_run(quil_program=quil_program,
+
+        self._memory_results = self.connection._qvm_run(quil_program=quil_program,
                                                         classical_addresses=classical_addresses,
                                                         trials=trials,
                                                         measurement_noise=self.measurement_noise,
                                                         gate_noise=self.gate_noise,
-                                                        random_seed=self.random_seed)['ro']
-        except KeyError:
-            warnings.warn("You are running a QVM program with no MEASURE instructions. "
-                          "The result of this program will always be an empty array. Are "
-                          "you sure you didn't mean to measure some of your qubits?")
-            self._bitstrings = np.zeros((trials, 0), dtype=np.int64)
+                                                        random_seed=self.random_seed)
+
+        if "ro" not in self._memory_results or self._memory_results["ro"] == []:
+            self._memory_results["ro"] = np.zeros((trials, 0))
```

Comment:
Don't know if it matters, but you dropped the warning and the `dtype` here. Can't imagine why it would matter what the element type of an array with no entries is but :man_shrugging:

Reply:
The warning was wrong anyway. An array of zeros is not empty.

Reply:
I put the dtype back but not the warning. Not putting things in `ro` is different from running a QVM program without MEASURE instructions — which one might want to do anyway with stateful QVMs!